### PR TITLE
fix(messaging): show the full message in previews & notifications

### PR DESCRIPTION
**Pull request by lilseyi**
## Summary

- Notification **emails** (chat-request, leader-DM, @mention) now quote the whole message — safety cap ~1000 chars, "…" only when the cap is hit (was: hard `slice(0,100)` with no ellipsis, cutting mid-word).
- **Push** bodies carry up to 200 chars and always end "…" when shortened (new optional `messagePushPreview` in the notification definitions; email keeps the full value).
- **Inbox / group-list rows** show up to 2 lines of the message preview (5 `numberOfLines` clamps changed; name/title clamps untouched); stored `lastMessagePreview` raised 100 → 200 so rows have content to fill.
- New `apps/convex/lib/textPreview.ts` — UTF-16-safe truncation (no split surrogate pairs / dangling ZWJ), used at every remaining cut point. No new dependencies.

## Acceptance criteria

From issue #661's "Done when" checklist ("Done when" is its criteria section):

- [x] Email quote box shows the whole message up to ~1000 chars, never a mid-word cut, "…" only at the cap — incl. a test using the reported "Please keep me in prayers" sentence.
- [x] Push shows noticeably more (200 chars via `truncateForBody`) and ends "…" when shortened.
- [x] Inbox/group rows show up to 2 lines ending "…"; row height is padding-driven so rows grow by one line, list layout unchanged.
- [x] Non-text messages keep "Sent a photo" / "Shared an event" / etc. in all three surfaces (canned-label branches untouched, test added).
- [x] Emoji at a cut boundary never renders broken — helper tested at push and stored-preview boundaries (🙏 across the cut; regex asserts no lone surrogate).
- [x] Group message, 1:1 DM request, and @mention email paths all read fully.

### Tests
```
apps/convex: npx vitest run __tests__/messaging __tests__/mention-email.test.ts
             __tests__/email-render.test.ts __tests__/moderation.test.ts
             __tests__/text-preview.test.ts
             -> 27 files, 661 tests passed (17 new/updated for this change)
typecheck:   apps/convex tsc clean; apps/mobile zero new errors in touched files
```

### Other consumers audited
SMS channel is unimplemented (`lib/notifications/send.ts:238`); event blasts and moderation notifications slice on their own separate paths (untouched); CLI channel list re-slices to 50 locally. `messagePushPreview` is optional so unupdated callers fall back to old behavior.

### Deploy note
Touches `apps/convex/**` **function logic only** — no schema, crons, or migrations. Merging deploys to staging immediately (no preview step); tests are green.

Known one-line follow-up (not in this PR): `apps/convex/schema.ts:1722` carries a stale `// First 100 chars` comment on `lastMessagePreview` — schema.ts is CODEOWNERS-protected, left for a human edit.

Closes #661

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/convex/lib/textPreview.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  return out;
+}
+
+/**
+ * Truncate to at most `max` UTF-16 units (ellipsis included in the budget),
+ * suffixing "…" only when the text was actually shortened.
+ */
+export function truncateWithEllipsis(text: string, max: number): string {
+  if (text.length <= max) return text;
+  return truncateChars(text, Math.max(0, max - 1)) + "…";
```
[the remark this answers is not in the document]
Comment (lilseyi) [reply apps/convex/lib/textPreview.ts]: Good catch — fixed in `9132e778`. `truncateWithEllipsis` now backs up to the last whitespace within 25% of the budget before appending the ellipsis, with the character-safe hard cut as the fallback so a single overlong token (URL, unspaced CJK) cannot shrink the preview to almost nothing. Tests added for the mid-word cut, the already-at-a-boundary case, the overlong-token fallback, and the budget invariant.
